src/coreclr/debug/daccess/cdac.cpp

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "cdac.h"
+#include <sospriv.h>
+#include <sstring.h>
+#include "dbgutil.h"
+
+#define CDAC_LIB_NAME MAKEDLLNAME_W(W("cdacreader"))
+
+namespace
+{
+    bool TryLoadCDACLibrary(HMODULE *phCDAC)
+    {
+        // Load cdacreader from next to DAC binary
+        PathString path;
+        if (FAILED(GetClrModuleDirectory(path)))
+            return false;
+
+        path.Append(CDAC_LIB_NAME);
+        *phCDAC = CLRLoadLibrary(path.GetUnicode());
```

Comment:
DAC can be downloaded from a symbol server into a symbol cache. Is it guarantee guaranteed that there cannot be a malicious copy of CDAC in the symbol cache next to it?

Reply:
I think that would be the correct solution. We'd need to check that if the DAC is signed with the special .NET Diagnostics cert, that the cDAC is also signed with that same cert.

Though that does mean for shipping, that we'll need to add the cDAC as another item that can be signed by this cert (it's currently strongly limited to the current DAC/DBI, but it should be possible to update the config).

Reply:
It would be tricky security-sensitive code.

Would it better to gate loading of CDAC behind an environment variable so that it is not loaded by default (for now at least)?

Reply:
> Do we already do a check for the special .NET Diagnostics cert anywhere that someone could point me at?

We have some examples in the sign-diagnostic-files.yml template (the PowerShell script does that). No examples in C or C++ though.